### DISCUSSION TITLE
vips_sharpen() requires VIPS_FORMAT_SHORT, but why does it not apply such cast itself?

### DISCUSSION THREAD

**Discussion by VivitionDeveloper** [libvips/libvips#4147] [General]
I just discovered that vips_sharpen requires band format VIPS_FORMAT_SHORT, but why does vips_sharpen itself not perform a conversion?

In my case, the saturation was adjusted by adjusting the AB values of LABS. As that correction was applied by VImage.linear({1, 1.1, 1.1}, 0) then the resulting LABS image became floats (I guess). When the sharpen was applied, an error was raised:
```
sharpen: image must be VIPS_FORMAT_SHORT
```
I found this check came from: https://github.com/libvips/libvips/blob/12cf4bd710acadfad221a0e1047936a30a47b4f6/libvips/convolution/sharpen.c#L203

My pseudo code is:
```
auto img = img.linear({1, 1.1, 1.1}, 0);
img = img.sharpen(VImage::option()->
					set("sigma", 3 / 4.0)->
					set("x1", 2)->
					set("y2", 10)->
					set("y3", 20)->
					set("m1", 0.5)->
					set("m2", 1));
```

I can of course apply a cast, but why does vips_sharpen itself not apply a cast for an image that is already in LABS color space?

**Comment by jcupitt** [comment]
Hi again, you're right, `sharpen` checks for LABS, but not that it's also int16. I'll add a cast.

**Comment by jcupitt** [comment]
.... I pushed the change to master. Thanks for reporting this.

**Reply by VivitionDeveloper** [reply]
Thanks a lot for this change.
